### PR TITLE
[Executor][Internal] Refine error message of input assignment parser

### PR DESCRIPTION
**Pull request by PeiwenGaoMS**
# Description

Refine error message of input assignment parser.

# All Promptflow Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](../CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by zzn2 — APPROVED** [review]

File: src/promptflow/promptflow/executor/_input_assignment_parser.py [hunk 1]
```diff
@@ -19,18 +20,44 @@ def parse_value(i: InputAssignment, nodes_outputs: dict, flow_inputs: dict):
     if i.value_type == InputValueType.FLOW_INPUT:
         if i.value not in flow_inputs:
             flow_input_keys = ", ".join(flow_inputs.keys()) if flow_inputs is not None else None
-            raise InputNotFound(message=f"{i.value} is not found from flow inputs '{flow_input_keys}'")
+            raise InputNotFound(
+                message_format=(
+                    "The input '{input_name}' is not found from flow inputs '{flow_input_keys}'. "
```
Comment (chw-microsoft) [inline src/promptflow/promptflow/executor/_input_assignment_parser.py:25]: 1)  I think you might consider create a template for this? is this case only happen for single node run? 


2) do we have yaml test case for this error ?

File: src/promptflow/promptflow/executor/_input_assignment_parser.py [hunk 2]
```diff
@@ -19,18 +20,44 @@ def parse_value(i: InputAssignment, nodes_outputs: dict, flow_inputs: dict):
...
+                flow_input_keys=flow_input_keys,
+            )
         return flow_inputs[i.value]
     if i.value_type == InputValueType.NODE_REFERENCE:
         if i.section != "output":
-            raise UnsupportedReference(f"Unsupported reference {i.serialize()}")
+            raise UnsupportedReference(
+                message_format=(
+                    "The section '{reference_section}' of reference is currently unsupported. "
+                    "Please specify the output part of the node '{reference_node_name}'."
```
Comment (chw-microsoft) [inline src/promptflow/promptflow/executor/_input_assignment_parser.py:37]: 1. this is to ask user to use 'output' section if they use node reference. it does not mean the output is missing from referenced node
2. do we have template for this?

File: src/promptflow/promptflow/executor/_input_assignment_parser.py [hunk 3]
```diff
@@ -19,18 +20,44 @@ def parse_value(i: InputAssignment, nodes_outputs: dict, flow_inputs: dict):
...
+                ),
+                reference_section=i.section,
+                reference_node_name=i.value,
+            )
         if i.value not in nodes_outputs:
             node_output_keys = ", ".join(nodes_outputs.keys()) if nodes_outputs is not None else None
             raise InputNotFoundFromAncestorNodeOutput(
-                message=f"{i.value} is not found from ancestor node output '{node_output_keys}'"
+                message_format=(
+                    "The input '{input_name}' is not found from ancestor node outputs '{node_output_keys}'. "
```
Comment (chw-microsoft) [inline src/promptflow/promptflow/executor/_input_assignment_parser.py:46]: 1. node_output_keys is list , we do not need single quote it.
2. any template

File: src/promptflow/promptflow/executor/_input_assignment_parser.py [hunk 4]
```diff
@@ -19,18 +20,44 @@ def parse_value(i: InputAssignment, nodes_outputs: dict, flow_inputs: dict):
...
+                ),
+                input_name=i.value,
+                node_output_keys=node_output_keys,
             )
         return parse_node_property(i.value, nodes_outputs[i.value], i.property)
-    raise NotImplementedError(f"The value type {i.value_type} cannot be parsed for input {i}")
+    raise NotSupported(
+        message_format=(
+            "The type '{input_type}' is currently unsupported. "
+            "Please choose from available types: '{supported_output_type}' and try again.",
```
Comment (chw-microsoft) [inline src/promptflow/promptflow/executor/_input_assignment_parser.py:56]: remove single quote as well.

File: src/promptflow/promptflow/executor/_input_assignment_parser.py [hunk 5]
```diff
@@ -49,13 +76,25 @@ def parse_node_property(node_name, node_val, property=""):
                 elif index.isdigit():
                     index = int(index)
                 else:
-                    raise InvalidReferenceProperty(f"Invalid index {index} when accessing property {property}")
+                    raise InvalidReferenceProperty(
+                        message_format=(
+                            "Invalid index '{index}' when accessing property '{property}' of the node '{node_name}'. "
```
Comment (chw-microsoft) [inline src/promptflow/promptflow/executor/_input_assignment_parser.py:81]: 1. needs template
2. this is only when output property value is array ? maybe add this part into the message

File: src/promptflow/promptflow/executor/_input_assignment_parser.py (hunk 2 above)
Comment (PeiwenGaoMS) [reply src/promptflow/promptflow/executor/_input_assignment_parser.py:39]: 1. yes
2. updated.

File: src/promptflow/promptflow/executor/_input_assignment_parser.py (hunk 3 above)
Comment (PeiwenGaoMS) [reply src/promptflow/promptflow/executor/_input_assignment_parser.py:49]: 1. it is string.
2. updated.

File: src/promptflow/promptflow/executor/_input_assignment_parser.py (hunk 5 above)
Comment (PeiwenGaoMS) [reply src/promptflow/promptflow/executor/_input_assignment_parser.py:86]: 1. updated.
2. I tried building a test case for this error but failed.

File: src/promptflow/promptflow/executor/_input_assignment_parser.py (hunk 1 above)
Comment (PeiwenGaoMS) [reply src/promptflow/promptflow/executor/_input_assignment_parser.py:26]: 1. updated.
2. added unit tests

File: src/promptflow/promptflow/executor/_input_assignment_parser.py (hunk 4 above)
Comment (PeiwenGaoMS) [reply src/promptflow/promptflow/executor/_input_assignment_parser.py]: removed.

File: src/promptflow/promptflow/executor/_input_assignment_parser.py (hunk 3 above)
Comment (PeiwenGaoMS) [reply src/promptflow/promptflow/executor/_input_assignment_parser.py]: 1. change it to list

**Review by zzn2 — APPROVED** [review]

**Review by chw-microsoft — APPROVED** [review]

**Pull request merged**